lib/core/lib/generators/ros/model/model_generator.rb

```diff
@@ -7,12 +7,29 @@ class ModelGenerator < Rails::Generators::NamedBase
     include GeneratorsHelper
 
     def create_files
-      create_file "app/models/#{name}.rb", <<~FILE
-        # frozen_string_literal: true
+      invoke(:model)
 
-        class Entity < #{parent_module}ApplicationRecord
-        end
-      FILE
+      gsub_created_file
+      insert_spec_default_template_into_file
+    end
+
+    private
+
+    def gsub_created_file
+      gsub_file(
+        "app/models/#{name}.rb",
+        'ApplicationRecord',
+        "#{parent_module}ApplicationRecord"
+      )
+    end
+
+    def insert_spec_default_template_into_file
```

Comment:
This is modifying the model's spec so it should be in a separate file from the model generator